manager/allocator/networkallocator/networkallocator.go

```diff
@@ -186,12 +187,28 @@ func (na *NetworkAllocator) ServiceAllocate(s *api.Service) (err error) {
 		return
 	}
 
-	// First allocate VIPs for all the pre-populated endpoint attachments
-	for _, eAttach := range s.Endpoint.VirtualIPs {
-		if err = na.allocateVIP(eAttach); err != nil {
-			return
+	// Allocate VIPs for all the pre-populated endpoint attachments
+	eVIPs := s.Endpoint.VirtualIPs[:0]
+	var match bool
+	for idx, eAttach := range s.Endpoint.VirtualIPs {
+		match = false
+		for _, nAttach := range s.Spec.Task.Networks {
+			if nAttach.Target == eAttach.NetworkID {
+				match = true
+				if err = na.allocateVIP(eAttach); err != nil {
+					return
```

Comment:
Unrelated to the change, but let's make this `return err`. Otherwise, if someone uses `err := ...` in one of the scopes this belongs to, the `err` variable being set isn't the return value, and the function would accidentally return nil.